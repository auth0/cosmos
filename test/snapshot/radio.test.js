import React from 'react'
import renderer from 'react-test-renderer'
import { Radio } from '@auth0/cosmos'
import 'jest-styled-components'

describe('Radio', () => {
  it('renders with automation attribute', () => {
    const tree = renderer
      .create(
        <Radio name="cosmos">
          <Radio.Option value="one">Option</Radio.Option>
        </Radio>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
