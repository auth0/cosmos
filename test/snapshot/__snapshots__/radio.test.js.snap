// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders with automation attribute 1`] = `
.c0 .c1 {
  display: table;
  margin-bottom: 16px;
}

.c0 .c1:last-child {
  margin: 0;
}

.c2 {
  position: relative;
  cursor: pointer;
  padding-left: 24px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
}

.c2 input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
}

.c2 .c3 {
  position: absolute;
  top: 50%;
  left: 0;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  height: 16px;
  width: 16px;
  background-color: #FFF;
  border: 1px solid #BEBEBE;
  box-shadow: inset 0 1px 2px 0 rgba(0,0,0,0.20);
  border-radius: 50%;
}

.c2:hover input ~ .c3 {
  box-shadow: inset 0 1px 4px 0 rgba(0,0,0,0.20);
}

.c2 input:checked ~ .c3 {
  background-color: #3B99FC;
  border: 1px solid #2C90FC;
}

.c2 .c3:after {
  content: '';
  position: absolute;
  display: none;
}

.c2 input:checked ~ .c3:after {
  display: block;
}

.c2 .c3:after {
  background-color: #FFF;
  width: 6px;
  height: 6px;
  border-radius: 50%;
  box-shadow: 0 1px 2px 0 rgba(0,0,0,0.20);
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

<div
  className="c0"
  data-cosmos-key="radio"
  name="cosmos"
>
  <label
    className="c1 c2"
  >
    <input
      checked={false}
      data-cosmos-key="radio.option"
      name="cosmos"
      pepe="test"
      readOnly={true}
      type="radio"
      value="one"
    />
    <span
      className="c3 "
    />
    <span
      className=""
    >
      Option
    </span>
  </label>
</div>
`;
